chaos/cluster/reboot_test.go

```diff
@@ -0,0 +1,185 @@
+package cluster
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/coreos/ktestutil/utils"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	// global client to be used by all tests.
+	client kubernetes.Interface
+)
+
+func TestMain(m *testing.M) {
+	var kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	client, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	exitCode := m.Run()
+	os.Exit(exitCode)
+}
+
+func TestMaxDisruption(t *testing.T) {
+	cluster, err := New(client,
+		WithMaxDisruption(1),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	doneCh := make(chan struct{})
+	go func() {
+		defer func() { doneCh <- struct{}{} }()
+		if err := cluster.RebootMasters(2 * time.Minute); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	hosts := hostsFromNodes(cluster.Masters)
+	sshClient := utils.MustNewSSHClient(&utils.SSHConfig{Timeout: 10 * time.Second})
+	if err := wait.PollInfinite(10*time.Second, func() (bool, error) {
+		select {
+		case <-doneCh:
+			return true, nil
+		default:
+		}
+
+		rebooting := 0
+		for _, host := range hosts {
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			_, _, err := sshClient.ExecWithCtx(ctx, host, "sudo systemctl is-active kubelet")
+			if err != nil {
+				rebooting++
+			}
+			cancel()
+		}
+		if rebooting > 1 {
+			return false, fmt.Errorf("more than 1 node rebooting")
+		}
+
+		return false, nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestRebootAll(t *testing.T) {
+	cluster, err := New(client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	go func() {
+		if err := cluster.RebootAll(2 * time.Minute); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	hosts := hostsFromNodes(cluster.Masters)
```

Comment:
it seems there are quite some code duplicates. can we make this checking a func?